src/frontend/apps/desk/src/features/mail-domains/access-management/components/ChooseRole.tsx

```diff
@@ -0,0 +1,66 @@
+import { Radio, RadioGroup } from '@openfun/cunningham-react';
+import { useTranslation } from 'react-i18next';
+
+import { Role } from '../../domains';
+
+interface ChooseRoleProps {
+  availableRoles: Role[];
+  currentRole: Role;
+  disabled: boolean;
+  setRole: (role: Role) => void;
+}
+
+export const ChooseRole = ({
+  availableRoles,
+  disabled,
+  currentRole,
+  setRole,
+}: ChooseRoleProps) => {
+  const { t } = useTranslation();
+  const rolesToDisplay = Array.from(new Set([currentRole, ...availableRoles]));
```

Comment:
As for this one, the currently connected user may actually have a role in the available roles.

For example, I am an owner (currentRole = OWNER), and with this privilege, I can assign a user the VIEWER, ADMIN, or OWNER role.

I'm not too fond of this set and would like to find a better design for these props. I probably do not need to use the currentRole prop but to show the default role of the currently connected user.

Reply:
It is necessary to create a default option for the current role. Otherwise, we can not show the currently connected user his role. I have to keep the set to ensure we have no double for the moment.